src/engine/soldier_auto_revive_system.cpp

```diff
@@ -43,35 +43,28 @@ void SoldierAutoReviveSystem::Update(double DeltaTime)
         for (core::ProgramState::ClientDatas_t::iterator i=mProgramState.mClientDatas.begin(), e=mProgramState.mClientDatas.end();i!=e;++i)
         {
             core::ClientData& clientData=*i;
-            Opt<Actor> player(mScene.GetActor((*i).mClientActorGUID));
-            if (!player.IsValid())
+            auto deathTimeIt = mTimesOfDeath.find( i->mClientActorGUID );
```

Comment:
This modification might not handle the usecase when a game is running, a new client connects. Fills the soldier_properties and presses ready. He might never be spawned.